jsonnet/components/admission-webhook.libsonnet

```diff
@@ -0,0 +1,213 @@
+                    },
+                    topologyKey: 'kubernetes.io/hostname',
+                  },
+                ],
+              },
+            },
+            securityContext: {},
+            priorityClassName: 'system-cluster-critical',
+            containers:
+              std.map(
+                function(c)
+                  if c.name == 'prometheus-operator-admission-webhook' then
+                    c {
+                      args+: [
+                        '--web.enable-tls=true',
+                        '--web.tls-cipher-suites=' + params.tlsCipherSuites,
+                        '--web.tls-min-version=VersionTLS12',
+                        '--web.cert-file=/etc/tls/private/tls.crt',
+                        '--web.key-file=/etc/tls/private/tls.key',
+                      ],
+                      livenessProbe: {
+                        httpGet: {
+                          path: '/healthz',
+                          port: 'https',
+                          scheme: 'HTTPS',
+                        },
+                      },
+                      readinessProbe: {
+                        httpGet: {
+                          path: '/healthz',
+                          port: 'https',
+                          scheme: 'HTTPS',
+                        },
+                      },
+                      securityContext: {},
+                      terminationMessagePolicy: 'FallbackToLogsOnError',
+                      volumeMounts+: [
+                        {
+                          mountPath: '/etc/tls/private',
+                          name: 'tls-certificates',
+                          readOnly: true,
+                        },
+                      ],
+                    }
+                  else
+                    c,
+                super.containers,
+              ),
+            volumes+: [
+              {
+                name: 'tls-certificates',
+                secret: {
+                  secretName: 'prometheus-operator-admission-webhook-tls',
+                  items: [
+                    {
+                      key: 'tls.crt',
+                      path: 'tls.crt',
+                    },
+                    {
+                      key: 'tls.key',
+                      path: 'tls.key',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        },
+      },
+    },
+
+    service+: {
+      metadata+: {
+        annotations+: {
+          'service.beta.openshift.io/serving-cert-secret-name': 'prometheus-operator-admission-webhook-tls',
+        },
+      },
+    },
+
+    // We collect no metrics from the admission webhook service.
+    // The availability of the service is measured at the Kubernetes API
+    // level using the apiserver_admission_webhook_* metrics available from
+    // the API server.
+    serviceMonitor:: {},
+
+    podDisruptionBudget: {
```

Comment:
Can we upstream PDB related additions?

Reply:
sounds good indeed. For now I've added a TODO comment but I'll carry this upstream.